src/components/BlogDeleteModal.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import { Modal, ModalBody, ModalFooter, ModalHeader, Button } from 'reactstrap';
+
+import { csrfFetch } from '../util/CSRF';
+
+class BlogDeleteModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: Boolean(props.isOpen),
+    };
+
+    this.close = this.close.bind(this);
```

Comment:
Instead of having close here, have close be a function on the context menu passed down to the modal. Have the isOpen state be managed by the context menu.